src/framework/dockwindow/internal/docktabsmodel.cpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "docktabsmodel.h"
+#include "log.h"
+
+using namespace muse::dock;
+
+DockTabsModel::DockTabsModel(QObject* parent)
+    : QAbstractListModel{parent}
+{
+}
+
+QVariant DockTabsModel::data(const QModelIndex& index, int role) const
+{
+    if (!index.isValid() || index.row() >= rowCount()) {
+        return QVariant();
+    }
+
+    const DockTab& tab = m_dockTabs.at(index.row());
+
+    switch (role) {
+    case TitleRole: return tab.title;
+    case ContextMenuRole: return tab.contextMenu;
+    case ToolBarComponentRole: return tab.toolBarComponent;
+    }
+
+    return QVariant();
+}
+
+QHash<int, QByteArray> DockTabsModel::roleNames() const
+{
+    static const QHash<int, QByteArray> roles = {
+        { TitleRole, "titleRole" },
+        { ContextMenuRole, "contextMenuRole" },
+        { ToolBarComponentRole, "toolBarComponentRole" },
```

Comment:
I would suggest not including the word `role` in the QML names anymore, because it reads a bit unnaturally IMO.